src/components/MainRouter/MainRouter.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import MainRouter from "./MainRouter";
+import { createMemoryHistory } from "history";
+import { Router } from "react-router";
+import AuthProvider from "../../components/AuthProvider";
+
+const renderComponent = (initialAuthState: boolean) => (
+  initialPath: string
+) => {
+  const history = createMemoryHistory({ initialEntries: [initialPath] });
+  return {
+    history,
+    ...render(
+      <AuthProvider initialAuthState={initialAuthState}>
+        <Router history={history}>
+          <MainRouter />
+        </Router>
+      </AuthProvider>
+    ),
+  };
+};
+
+// TODO extend jest matchers instead
+const expectToBeOnPage = (
+  container: HTMLElement,
+  myPath: string,
+  expectedPage: string
+) => {
+  const expectedPath = `/${expectedPage}`;
+  expect(myPath).toEqual(expectedPath);
+  expect(container.querySelector(".ion-page")).toHaveAttribute(
+    "title",
+    expectedPage
+  );
+};
+
+describe("Unauthenticated User", () => {
+  const renderUnauthenticatedComponent = renderComponent(false);
+
+  [
+    { initialPath: "/home", expectedPage: "login" },
+    { initialPath: "/home/1234", expectedPage: "login" },
+    { initialPath: "/login", expectedPage: "login" },
+    { initialPath: "/login/1234", expectedPage: "login" },
+    { initialPath: "/", expectedPage: "login" },
+    { initialPath: "/bad-route", expectedPage: "login" },
```

Comment:
do we want a 404 page ? 🤔  (just thinking out loud)

Reply:
yeah, its nice to have one.

Reply:
@patrixr @bharat619 any reason for 404 page? Current implementation makes it impossible to reach such a page. What are the possible disadvantages of this?

Reply:
I'm talking about future implementation, not current.
and I just like 404s